architecture/2024-05-radius-on-dapr.md

```diff
@@ -0,0 +1,324 @@
+| ------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Building Block API | Distributed systems primitives provided by Dapr. Each building block has multiple components that implement it. Examples include state store, pub/sub, workflows. [docs](https://docs.dapr.io/developing-applications/building-blocks/).                                                                                                                              |
+| Component          | An implementation of a specific building block. For example, Dapr provides a Redis component that implements the state store building block. Component is also used as the term for Dapr's configuration format.                                                                                                                                                      |
+| Pub/Sub            | A messaging building-block that implements commonly-used [pub/sub](https://aws.amazon.com/what-is/pub-sub-messaging/#:~:text=The%20publish%2Dsubscribe%20(pub%2Fsub)%20model%20enables%20event,placed%20in%20a%20shopping%20cart.) functionality. See [docs](https://docs.dapr.io/reference/components-reference/supported-pubsub/) for supported pub/sub components. |
+| State store        | A key-value-store building block. See [docs](https://docs.dapr.io/reference/components-reference/supported-state-stores/) for supported state stores.                                                                                                                                                                                                                 |
+| Workflows          | A [workflow-as-code](https://docs.dapr.io/developing-applications/building-blocks/workflow/workflow-overview/) programming model for reliable distributed applications.                                                                                                                                                                                               |
+
+
+## Objectives
+
+> **Issue Reference:** <!-- (If appropriate) Reference an existing issue that describes the feature or bug. -->
+
+### Goals
+
+- Adopt Dapr for our data storage and messaging needs. 
+  - Keep `rad init` and `rad install` simple by including Dapr and a default choice of components as part of our Helm chart.
```

Comment:
What would happen if my cluster already has **Dapr** installed?

Reply:
We can talk about this question above. Would there be a clash? Or are two **Dapr**s (one inside Radius and one running on the cluster) two different things?

Reply:
I think we'd skip it. We already have logic for this for Contour. We should be consistent with that behavior unless we have a reason not to.